lib/imagec/download.go

```diff
@@ -154,15 +151,16 @@ func (ldm *LayerDownloader) DownloadLayers(ctx context.Context, ic *ImageC) erro
 				defer topDownload.Transfer.Release(watcher)
 
 				ldm.registerDownload(topDownload)
-				cache.LayerCache().AddNew(id)
+				layer.Downloading = true
+				LayerCache().Add(layer)
 
 				continue
 			default:
 				return err
 			}
 		}
 
-		if downloading {
+		if layerConfig.Downloading {
```

Comment:
Right now, code that want's to change `layerConfig.Downloading` calls `LayerCache().Commit()` which uses a lock, but there's nothing to stop code from changing this value manually. I think it's better not to export it, but instead expose a getter/setter in the form of `IsDownloading()` and `Commit()`. I will push a change.